### PR TITLE
Fix PHP notice in AdSense constructor by removing additional unused datapoints

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #2507

## Relevant technical choices

* Fixes initial follow-up issue related to PHP notice raised during `setup_info` call by inlining the fallback case that we were already using for the module's `homeurl`
* `GET:reports-url` datapoint is no longer used and removed
* Initially extracted `GET:account-url` datapoint to a method for it to be used in `GET:notifications` so the now unused `GET:account-url` could be removed
* ~~Found that AdSense's `GET:notifications` datapoint is also unused and removed this as well, which allowed for the added method for `get_account_url` to also be removed as it was now unused as well~~
    - ~~I was a bit surprised about this but after looking through the code, the AdSense module uses `getAlerts`, never `getNotifications` and all references to the notifications datapoint are all for site notifications, no module notifications are ever requested. Even the datapoint for AdSense's `GET:notifications` used `get_data( 'alerts' )` internally so it makes sense that this would no longer be used. This should simplify the changes needed in #2468 as well.~~

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Modules/AdSense.php [hunk 1]
```diff
@@ -123,21 +123,6 @@ public function prepare_info_for_js() {
...
-		);
-
-		$signup_args = array(
-			'utm_source' => 'site-kit',
-			'utm_medium' => 'wordpress_signup',
-		);
-
-		$info['accountURL'] = add_query_arg( $idenfifier_args, $this->get_data( 'account-url' ) );
-		$info['signupURL']  = add_query_arg( $signup_args, $info['accountURL'] );
-		$info['rootURL']    = add_query_arg( $idenfifier_args, 'https://www.google.com/adsense/' );
```
Comment (aaemnnosttv) [inline includes/Modules/AdSense.php:139]: These are all unused now. We likely missed this cleanup when we added the `getServiceURL` selectors for AdSense.

File: includes/Modules/AdSense.php [hunk 2]
```diff
@@ -310,44 +279,6 @@ protected function create_data_request( Data_Request $data ) {
 				}
 
 				return $this->create_adsense_earning_data_request( array_filter( $args ) );
-			case 'GET:notifications':
-				return function() {
-					$alerts = $this->get_data( 'alerts' );
```
Comment (aaemnnosttv) [inline includes/Modules/AdSense.php:315]: AdSense uses `getAlerts` directly which uses `GET:alerts` so this is no longer used.

**Review by eugene-manuilov — DISMISSED** [review]
Awesome! Thanks, @aaemnnosttv!

File: includes/Modules/AdSense.php (hunk 2 above)
Comment (felixarntz) [reply includes/Modules/AdSense.php:315]: @aaemnnosttv This must not be removed. `getAlerts` is used during setup, but we still have legacy notifications functionality relying on this. The overall `GET:modules/adsense/notifications` endpoint will look if this particular datapoint here exists and use it, so this block needs to be kept in place.

**Review by felixarntz — CHANGES_REQUESTED** [review]
One thing here shouldn't be removed, otherwise LGTM.

File: includes/Modules/AdSense.php (hunk 2 above)
Comment (aaemnnosttv) [reply includes/Modules/AdSense.php:315]: @felixarntz I know it's connected to that endpoint but I couldn't find where it's actually used. Can you point me to it?

Comment (felixarntz) [reply includes/Modules/AdSense.php:315]: @aaemnnosttv It's used in the `Alert` component which is included on every module page for the respective module. Basically if some alert comes up in your AdSense account, you'll see it on the AdSense module page.

Comment (aaemnnosttv) [reply includes/Modules/AdSense.php:315]: Ah, I see it now. Apparently also used in `DashboardModulesAlerts`. Thanks!

Comment (aaemnnosttv) [reply includes/Modules/AdSense.php]: Restored 👍

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**